packages/material-ui-system/src/style.js

```diff
@@ -14,7 +14,7 @@ function style(options) {
 
   const fn = props => {
     if (props[prop] == null) {
-      return null;
+      return {};
```

Comment:
Does this change impact performance? Let's see, I'm running the perf benchmark.

Reply:
Before
- @material-ui/core all-inclusive x 43,602 ops/sec ±1.30% (187 runs sampled)
- @material-ui/core all-inclusive x 42,558 ops/sec ±3.54% (190 runs sampled)

After
- @material-ui/core all-inclusive x 26,452 ops/sec ±6.68% (176 runs sampled)
- @material-ui/core all-inclusive x 28,100 ops/sec ±4.41% (176 runs sampled)

Reply:
Let's put it in perspective with styled-components. Hold on.

Reply:
I can't see any significant diff with styled-components. The system is not the main bottleneck.

Reply:
I can't see a small decrease with emotion:

**Before**
- Box emotion x 21,055 ops/sec ±5.07% (185 runs sampled)
- Box emotion x 20,870 ops/sec ±5.72% (182 runs sampled)

**After**
- Box emotion x 19,333 ops/sec ±1.32% (188 runs sampled)
- Box emotion x 19,224 ops/sec ±2.56% (187 runs sampled)

What alternative do we have? Part of the issue comes from the fact that we run all the style functions while we could be smarter about it, only run the ones that match an incoming prop.

Reply:
Of other CSS-in-JS libraries work well with null value, we could alter own function which do not (I think its makeStyle at least) to handle null correctly instead.

Reply:
@oliviertassinari Verified: Emotion and styled-components have no problem with `null` returned from style functions: https://codesandbox.io/s/condescending-leftpad-d4mr0

Reply:
@oliviertassinari I was looking through, to determine where the responsibility for the fix is best placed. It first seemed like `styled` should intercept and pass the values correctly to `makeStyles`. 

Then It rather looked like a general `makeStyles` error: passing it `{root: () => null}` cause same error, which obviously using `styled-components` and `emotion` does not cause the same error.
While it was clear to me, that the error is raised outside material-ui code, I have seen responsibility in some of the material-ui functions named above, because other CSS-in-JS libs handle it.

Now I realized its actually an issue where the error is raised, in the camelCase JSS plugin. The reason why styled-components and emotion have no problem is because they do not use JSS and therefore do not use this plugin... 

The plugin simply does not properly test style for it being actually and object. However checking the repo I found this [plugin](https://github.com/cssinjs/jss-camel-case) has been archived.

Reply:
Simply put: null can not be returned due to the camelCase plugin testing incorrectly. Therefore the approach in `[system] style` was almost correct. However it fixes it wrongly because the fix is not needed for other CSS-in-JS libraries as they do not have this faulty plugin ;)

Also the fix is not handling cases where you just use `makeStyle` and a function returns null, or where you use `styled` and provide a style function not creating using `[system] style`.

The fix INSIDE `mui` code would be for example wrapping every single functions in `[styles] getDynamicStyles` with a function that tests for `null` and returns `{}`, which kind of smells. 

... Or look for an alternative to the JSS plugin

Reply:
I think I hit the wrong version of the plugin. The archived one is the one without `plugin` in its name.
@oliviertassinari looks like similar to this [bug](https://github.com/cssinjs/jss/issues/489), the difference is that now its function returning null (or undefined)

Reply:
https://github.com/cssinjs/jss/issues/1159

Reply:
Looks like a bug in an alpha version of jss. Is there a reason you use the alpha version? Some important feature or bugfix there?